File: gluecodium/src/test/resources/smoke/instances/output/android/com/example/smoke/StructWithInterface.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
public final class StructWithInterface {
    @NonNull
    public SimpleInterface interfaceInstance;
    public StructWithInterface(@NonNull final SimpleInterface interfaceInstance) {
        this.interfaceInstance = interfaceInstance;
    }
}

File: gluecodium/src/test/resources/smoke/instances/output/cpp/src/smoke/StructWithInterface.cpp
// -------------------------------------------------------------------------------------------------
//
//
// -------------------------------------------------------------------------------------------------
#include "smoke/SimpleInterface.h"
#include "smoke/StructWithInterface.h"
#include <utility>
namespace smoke {
StructWithInterface::StructWithInterface( )
    : interface_instance{ }
{
}
StructWithInterface::StructWithInterface( ::std::shared_ptr< ::smoke::SimpleInterface > interface_instance )
    : interface_instance( std::move( interface_instance ) )
{
}
}
